pkg/resourcemanager/webhook/nodeagentauthorizer/authorizer.go

```diff
@@ -0,0 +1,233 @@
+	if allowed, reason := a.checkVerb(log, attrs, allowedVerbs...); !allowed {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	// Listing nodes must be allowed unconditionally, because gardener-node-agent only knows its hostname, but not its node. Kubelet
+	// creates the latter when gardener-node-agent is already running.
+	if attrs.GetVerb() == "list" || attrs.GetVerb() == "watch" {
+		return auth.DecisionAllow, "", nil
+	}
+
+	machine := &machinev1alpha1.Machine{}
+	if err := a.sourceClient.Get(ctx, client.ObjectKey{Name: machineName, Namespace: a.machineNamespace}, machine); err != nil {
+		return auth.DecisionNoOpinion, "", err
+	}
+
+	if machine.Labels[machinev1alpha1.NodeLabelKey] != attrs.GetName() {
+		log.Info("Denying authorization for node because it belongs to a different machine", "node", attrs.GetName(), "machine", machineName)
+		return auth.DecisionNoOpinion, fmt.Sprintf("node %q belongs does not belong to machine %q", attrs.GetName(), machineName), nil
+	}
+
+	return auth.DecisionAllow, "", nil
+}
+
+func (a *authorizer) authorizeSecret(ctx context.Context, log logr.Logger, machineName string, attrs auth.Attributes) (auth.Decision, string, error) {
+	if ok, reason := a.checkSubresource(log, attrs); !ok {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	allowedVerbs := []string{"get", "list", "watch"}
+	if allowed, reason := a.checkVerb(log, attrs, allowedVerbs...); !allowed {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	machine := &machinev1alpha1.Machine{}
+	if err := a.sourceClient.Get(ctx, client.ObjectKey{Name: machineName, Namespace: a.machineNamespace}, machine); err != nil {
+		return auth.DecisionNoOpinion, "", err
+	}
+
+	validSecrets := []string{machine.Spec.NodeTemplateSpec.Labels[v1beta1constants.LabelWorkerPoolGardenerNodeAgentSecretName], valitailTokenSecretName}
+
+	if !slices.Contains(validSecrets, attrs.GetName()) || attrs.GetNamespace() != metav1.NamespaceSystem {
+		log.Info("Denying authorization because gardener-node-agent is not allowed to access secret", "name", attrs.GetName(), "namespace", attrs.GetNamespace())
+		return auth.DecisionNoOpinion, fmt.Sprintf("gardener-node-agent can only access secrets %v in \"kube-system\" namespace", validSecrets), nil
+	}
+
+	return auth.DecisionAllow, "", nil
+}
+
+func (a *authorizer) checkVerb(log logr.Logger, attrs auth.Attributes, allowedVerbs ...string) (bool, string) {
+	if !slices.Contains(allowedVerbs, attrs.GetVerb()) {
+		log.Info("Denying authorization because verb is not allowed for this resource type", "allowedVerbs", allowedVerbs)
+		return false, fmt.Sprintf("only the following verbs are allowed for this resource type: %+v", allowedVerbs)
+	}
+
+	return true, ""
+}
+
+func (a *authorizer) checkSubresource(log logr.Logger, attrs auth.Attributes, allowedSubresources ...string) (bool, string) {
+	if subresource := attrs.GetSubresource(); len(subresource) > 0 && !slices.Contains(allowedSubresources, attrs.GetSubresource()) {
```

Comment:
```suggestion
	if subresource := attrs.GetSubresource(); len(subresource) > 0 && !slices.Contains(allowedSubresources, subresource) {
```

You could also fix it in the same func that is used for the Seed authorizer.